core/src/main/java/org/apache/carbondata/core/datastore/columnar/ByteArrayColumnWithRowId.java

```diff
@@ -51,11 +59,12 @@ public boolean equals(Object obj) {
       return false;
     }
     ByteArrayColumnWithRowId o = (ByteArrayColumnWithRowId)obj;
-    return Arrays.equals(column, o.column) && rowId == o.rowId;
+    return Arrays.equals(column, o.column) && rowId == o.rowId
+        && isNoDictionary == o.isNoDictionary;
```

Comment:
Do we really need to consider `isNoDictionary` here ? when it will be not equal? I think as it is a comparison of data within a column, this condition might always be true. 

If not required, remove from equals and hashcode

Reply:
Done. Removed.